lib/exercises.rb

```diff
@@ -6,17 +6,88 @@
 # Space Complexity: ?
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  return [] if strings.length == 0
+
+  sorted_strings = strings.map { |string|
+    string.chars.sort.join
+  }
+
+  sorted_strings_hash = {}
+
+  sorted_strings.each { |string|
+    sorted_strings_hash[string] = []
+  }
+
+  sorted_strings.each_with_index { |string, index|
+    sorted_strings_hash[string] << strings[index]
+  }
```

Comment:
Just noting the variable names here are a bit confusing string, strings, sorted strings hash etc.